### PR TITLE
Add task-topology plugin for scheduling.

### PR DESCRIPTION
**Pull request by jiangkaihua**

### CODE REVIEW

**Comment by k82cn** [comment]
please help to draft an design doc for this PR :)

File: pkg/scheduler/plugins/task-topology/bucket.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+type reqAction int
+
+const (
+	reqSub reqAction = iota
+	reqAdd
+)
+
+// Bucket is struct used to classify tasks by affinity and anti-affinity
+type Bucket struct {
```
Comment (Thor-wl) [inline pkg/scheduler/plugins/task-topology/bucket.go:33]: Is there a design doc and recommanded user guidance?

File: pkg/scheduler/plugins/task-topology/bucket.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		score += request
+	}
+
+	switch action {
+	case reqSub:
+		b.reqScore -= score
+		b.request.Sub(req)
+	case reqAdd:
+		b.reqScore += score
+		b.request.Add(req)
```
Comment (Thor-wl) [inline pkg/scheduler/plugins/task-topology/bucket.go:82]: Suggest to add default case

**Review by william-wang — APPROVED** [review]

**Comment by william-wang** [comment]
Please update the api dependence from volcano.sh/pkg/api to volcano.sh/api

**Review by william-wang — APPROVED** [review]
/lgtm

**Pull request merged**